openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -3,113 +3,90 @@
-            ...
-
-    Note: this should eventually be deprecated in favour of a dedicated `override_waffle_switch` context manager.
+    Deprecated class: instead, use edx_toggles.toggles.WaffleSwitchNamespace.
     """
 
     @contextmanager
     def override(self, switch_name, active=True):
         """
-        Overrides the active value for the given switch for the duration of this
-        contextmanager.
-        Note: The value is overridden in the request cache AND in the model.
+        Deprecated method: instead, use edx_toggles.toggles.testutils.override_waffle_switch.
         """
-        previous_active = self.is_enabled(switch_name)
-        try:
-            self.override_for_request(switch_name, active)
-            with self.override_in_model(switch_name, active):
-                yield
-        finally:
-            self.override_for_request(switch_name, previous_active)
-
-    def override_for_request(self, switch_name, active=True):
-        """
-        Overrides the active value for the given switch for the remainder of
-        this request (as this is not a context manager).
-        Note: The value is overridden in the request cache, not in the model.
-        """
-        namespaced_switch_name = self._namespaced_name(switch_name)
-        self._cached_switches[namespaced_switch_name] = active
-        log.info(
-            "%sSwitch '%s' set to %s for request.",
-            self.log_prefix,
-            namespaced_switch_name,
-            active,
+        warnings.warn(
+            (
+                "WaffleSwitchNamespace.override is deprecated. Instead, use"
+                " edx_toggles.toggles.testutils.override_waffle_switch."
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        set_custom_attribute(
+            "deprecated_waffle_utils", "WaffleSwitchNamespace.override"
         )
+        from edx_toggles.toggles.testutils import override_waffle_switch
 
-    @contextmanager
-    def override_in_model(self, switch_name, active=True):
-        """
-        Overrides the active value for the given switch for the duration of this
-        contextmanager.
-        Note: The value is overridden in the model, not the request cache.
-        Note: This should probably be moved to a test class.
-        """
-        # Import is placed here to avoid model import at project startup.
-        # pylint: disable=import-outside-toplevel
-        from waffle.testutils import override_switch as waffle_override_switch
-
-        namespaced_switch_name = self._namespaced_name(switch_name)
-        with waffle_override_switch(namespaced_switch_name, active):
-            log.info(
-                "%sSwitch '%s' set to %s in model.",
-                self.log_prefix,
-                namespaced_switch_name,
-                active,
-            )
+        with override_waffle_switch(BaseWaffleSwitch(self, switch_name), active):
```

Comment:
This PR is no longer using this version of `WaffleSwitchNamespace`, right? 
Should we separate this in a different PR, or can you ensure that locally you can use this somewhere and the test passes?

Reply:
> This PR is no longer using this version of WaffleSwitchNamespace, right?

Yes, this is correct.

> Should we separate this in a different PR, or can you ensure that locally you can use this somewhere and the test passes?

I'll add a unit test.